core/src/main/java/com/amplifyframework/predictions/models/EntityDetails.java

```diff
@@ -0,0 +1,291 @@
+ */
+
+package com.amplifyframework.predictions.models;
+
+import android.graphics.RectF;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.amplifyframework.util.Immutable;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Holds the entity detection results
+ * for the predictions category.
+ */
+public final class EntityDetails {
+    private final RectF box;
+    private final Polygon polygon;
+    private final AgeRange ageRange;
+    private final Pose pose;
+    private final Gender gender;
+    private final List<Landmark> landmarks;
+    private final List<Emotion> emotions;
+    private final List<BinaryFeature> features;
+
+    private EntityDetails(final Builder builder) {
+        this.box = builder.getBox();
+        this.polygon = builder.getPolygon();
+        this.ageRange = builder.getAgeRange();
+        this.pose = builder.getPose();
+        this.gender = builder.getGender();
+        this.landmarks = builder.getLandmarks();
+        this.emotions = builder.getEmotions();
+        this.features = builder.getValues();
+    }
+
+    /**
+     * Gets the rectangular target boundary if available.
+     * @return the rectangular boundary
+     */
+    @Nullable
+    public RectF getBox() {
+        return box;
+    }
+
+    /**
+     * Gets a more finely defined target boundary if available.
+     * @return the polygonal boundary
+     */
+    @Nullable
+    public Polygon getPolygon() {
+        return polygon;
+    }
+
+    /**
+     * Gets the range of possible ages.
+     * @return the range ages
+     */
+    @Nullable
+    public AgeRange getAgeRange() {
+        return ageRange;
+    }
+
+    /**
+     * Gets the detected pose.
+     * @return the pose feature
+     */
+    @Nullable
+    public Pose getPose() {
+        return pose;
+    }
+
+    /**
+     * Gets the detected gender.
+     * @return the gender type feature
+     */
+    @Nullable
+    public Gender getGender() {
+        return gender;
+    }
+
+    /**
+     * Gets the list of detected landmarks.
+     * @return the list of landmarks
+     */
+    @NonNull
+    public List<Landmark> getLandmarks() {
+        return Immutable.of(landmarks);
+    }
+
+    /**
+     * Gets the list of detected emotions.
+     * @return the list of emotions
+     */
+    @NonNull
+    public List<Emotion> getEmotions() {
+        return Immutable.of(emotions);
+    }
+
+    /**
+     * Gets the list of boolean features.
+     * @return the list of features
+     */
+    @NonNull
+    public List<BinaryFeature> getValues() {
+        return Immutable.of(features);
+    }
+
+    /**
+     * Gets a builder for entity detection result.
+     * @return a new builder
+     */
+    @NonNull
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Builder for {@link EntityDetails}.
+     */
+    public static final class Builder {
```

Comment:
In this builder, you permit a number of `@Nullable` options. What is their purpose? If a user sets a field to `null`, does it clear it? And mean "I don't know"? Or, what are the implications of leaving these things unspecified? When can an `EntityDetails` have different collections of these various properties?

Reply:
It means that some of the properties that our implementation chose to detect are not required for other plugins to detect. So, a `null` would mean "I do not detect this value in my implementation of `identify`".